src/test/rustdoc-ui/check-doc-alias-attr-location.rs

```diff
@@ -0,0 +1,23 @@
+#![feature(doc_alias)]
+
+pub struct Bar;
+pub trait Foo {
+    type X;
+    fn foo() -> Self::X;
+}
+
+#[doc(alias = "foo")] //~ ERROR
+extern {}
+
+#[doc(alias = "bar")] //~ ERROR
+impl Bar {
+    #[doc(alias = "const")] //~ ERROR
```

Comment:
This shouldn't be an error. It's only `const`s in trait impls that don't appear in the search index. This one does appear.

Reply:
However this one doesn't appear in the search.

Reply:
Associated consts on inherent impls do appear in the search index. It's only ones from trait impls that don't. The particular one isn't marked `pub` so it's missing unless you pass `--document-private-items` but the error message shouldn't consider that.

Reply:
Oh damn, I thought it was like traits and that it was public by default. My bad...